test/Microsoft.VisualStudio.Threading.Analyzers.Tests/VSTHRD004AwaitSwitchToMainThreadAsyncAnalyzerTests.cs

```diff
@@ -104,6 +104,26 @@ async Task FooAsync()
         await CSVerify.VerifyAnalyzerAsync(test);
     }
 
+    [Fact]
+    public async Task AsyncMethodWithAwaitNoThrowAwaitable_ProducesNoDiagnostic()
+    {
+        var test = @"
+using System.Threading.Tasks;
+
+class Test
+{
+    Microsoft.VisualStudio.Threading.JoinableTaskFactory jtf;
+
+    async Task FooAsync()
+    {
+        await jtf.SwitchToMainThreadAsync().NoThrowAwaitable();
+    }
+}
+";
+
```

Comment:
nit: extra newline.

Reply:
An empty line between the test snippet and the assertion is actually customary in this repo.